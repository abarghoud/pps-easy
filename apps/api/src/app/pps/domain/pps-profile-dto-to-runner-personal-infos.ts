import { PPSProfileDto } from './pps-profile-dto.model';
import { Gender } from '../../gender.enum';
import { DateTime } from 'luxon';

export class PPSProfileDTOToRunnerPersonalInfos {
  public readonly birthdayDay: number;
  public readonly birthdayMonth: number;
  public readonly birthdayYear: number;
  public readonly email: string;
  public readonly firstname: string;
  public readonly gender: Gender;
  public readonly lastname: string;
  public readonly eventDate: DateTime;

  public constructor(ppsDto: PPSProfileDto) {
    const luxonBirthday = DateTime.fromJSDate(ppsDto.birthday);
    this.birthdayDay = luxonBirthday.day;
    this.birthdayMonth = luxonBirthday.month;
    this.birthdayYear = luxonBirthday.year;
    this.email = ppsDto.email;
    this.firstname = ppsDto.firstname;
    this.gender = ppsDto.gender;
    this.lastname = ppsDto.lastname;
    this.eventDate = DateTime.fromJSDate(ppsDto.event_date);
  }
}
