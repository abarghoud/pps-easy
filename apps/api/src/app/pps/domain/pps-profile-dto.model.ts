import { IsDateString, IsEmail, IsEnum, IsString } from 'class-validator';

import { Gender } from '../../gender.enum';

export class PPSProfileDto {
  @IsDateString()
  public birthday: string;

  @IsEmail()
  public email: string;

  @IsDateString()
  public event_date: string;

  @IsString()
  public firstname: string;

  @IsEnum(Gender)
  public gender: Gender;

  @IsString()
  public lastname: string;
}
