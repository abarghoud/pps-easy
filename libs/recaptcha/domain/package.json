{
  "name": "@pps-easy/recaptcha/domain",
  "version": "0.0.1",
  "dependencies": {},
  "private": true
}
