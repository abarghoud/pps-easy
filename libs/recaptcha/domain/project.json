{
  "name": "domain",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/recaptcha/domain/src",
  "projectType": "library",
  "tags": [],
  "// targets": "to see all targets run: nx show project domain --web",
  "targets": {}
}
