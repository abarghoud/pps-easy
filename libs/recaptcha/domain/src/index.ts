export * from './lib/challenge-result';
