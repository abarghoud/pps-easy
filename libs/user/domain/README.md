# user-domain

This library was generated with [Nx](https://nx.dev).

## Building

Run `nx build user-domain` to build the library.

## Running unit tests

Run `nx test user-domain` to execute the unit tests via [Vitest](https://vitest.dev/).
