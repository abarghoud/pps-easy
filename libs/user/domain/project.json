{
  "name": "user-domain",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/user/domain/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/user/domain",
        "main": "libs/user/domain/src/index.ts",
        "tsConfig": "libs/user/domain/tsconfig.lib.json",
        "assets": ["libs/user/domain/*.md"]
      }
    }
  }
}
