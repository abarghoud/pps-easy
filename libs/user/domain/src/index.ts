export * from './lib/user-personal-info-entity';
export * from './lib/gender.enum';
