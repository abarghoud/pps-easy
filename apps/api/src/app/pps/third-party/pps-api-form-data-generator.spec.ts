import { Gender } from '@pps-easy/user/domain';

import { getDefaultPPSFormData, PPSApiFormDataGenerator } from './pps-api-form-data-generator';
import { PPSProfileDTOToRunnerPersonalInfos } from '../domain/pps-profile-dto-to-runner-personal-infos';
import { PPSProfileDto } from '../domain/pps-profile-dto.model';


const getFakePPSDto = (): PPSProfileDto => {
  const ppsDto = new PPSProfileDto();

  ppsDto.event_date = new Date('2024-12-31');
  ppsDto.birthday = new Date('1990-01-01');
  ppsDto.email = 'test@test.com';
  ppsDto.gender = Gender.male;
  ppsDto.lastname = 'DUPONT';
  ppsDto.firstname = 'Marc';

  return ppsDto;
}

describe('The PPSApiFormDataGenerator class', () => {
  let ppsApiFormDataGenerator: PPSApiFormDataGenerator;
  const ppsProfileInfos = new PPSProfileDTOToRunnerPersonalInfos(getFakePPSDto());

  beforeEach(() => {
    ppsApiFormDataGenerator = new PPSApiFormDataGenerator(ppsProfileInfos);
  });

  describe('The generateRaceDateFormData method', () => {
    it('should return a formData containing the default values with provided authenticity_token and race date', () => {
      const authenticityToken = 'the_token';
      const expectedFormData = getDefaultPPSFormData(authenticityToken);

      expectedFormData.append('course[race_date]', ppsProfileInfos.eventDate.toFormat('dd-MM-yyyy'));

      expect(ppsApiFormDataGenerator.generateEventDateFormData(authenticityToken)).toEqual(expectedFormData);
    });
  });

  describe('The generatePersonalInfoFormData method', () => {
    it('should return a formData containing the default values with provided authenticity_token and personal info', () => {
      const authenticityToken = 'the_token';

      expect(ppsApiFormDataGenerator.generatePersonalInfoFormData(authenticityToken)).toMatchSnapshot();
    });
  });

  describe('The generateCardiovascularRisksFormData method', () => {
    it('should return a formData containing the default values with provided authenticity_token and personal info', () => {
      const authenticityToken = 'the_token';

      expect(ppsApiFormDataGenerator.generateCardiovascularRisksFormData(authenticityToken)).toMatchSnapshot();
    });
  });

  describe('The generateRiskFactorsFormData method', () => {
    it('should return a formData containing the default values with provided authenticity_token and personal info', () => {
      const authenticityToken = 'the_token';

      expect(ppsApiFormDataGenerator.generateRiskFactorsFormData(authenticityToken)).toMatchSnapshot();
    });
  });

  describe('The generatePrecautionsFormData method', () => {
    it('should return a formData containing the default values with provided authenticity_token and personal info', () => {
      const authenticityToken = 'the_token';

      expect(ppsApiFormDataGenerator.generatePrecautionsFormData(authenticityToken)).toMatchSnapshot();
    });
  });

  describe('The generateFinalizationFormData method', () => {
    it('should return a formData containing the default values with provided authenticity_token and personal info', () => {
      const authenticityToken = 'the_token';

      expect(ppsApiFormDataGenerator.generateFinalizationFormData(authenticityToken)).toMatchSnapshot();
    });
  });
});
