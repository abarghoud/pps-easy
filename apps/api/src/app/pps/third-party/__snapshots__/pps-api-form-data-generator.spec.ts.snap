// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The PPSApiFormDataGenerator class The generateCardiovascularRisksFormData method should return a formData containing the default values with provided authenticity_token and personal info 1`] = `
FormData {
  Symbol(state): [
    {
      "name": "authenticity_token",
      "value": "the_token",
    },
    {
      "name": "button",
      "value": "",
    },
    {
      "name": "_method",
      "value": "patch",
    },
    {
      "name": "course[cardiovascular_risks_video]",
      "value": "1",
    },
    {
      "name": "course[cardiovascular_risks_checkbox]",
      "value": "1",
    },
  ],
}
`;

exports[`The PPSApiFormDataGenerator class The generateFinalizationFormData method should return a formData containing the default values with provided authenticity_token and personal info 1`] = `
FormData {
  Symbol(state): [
    {
      "name": "authenticity_token",
      "value": "the_token",
    },
    {
      "name": "button",
      "value": "",
    },
    {
      "name": "_method",
      "value": "patch",
    },
    {
      "name": "course[finalization_checkbox]",
      "value": "1",
    },
    {
      "name": "course[ffa_newsletter]",
      "value": "0",
    },
  ],
}
`;

exports[`The PPSApiFormDataGenerator class The generatePersonalInfoFormData method should return a formData containing the default values with provided authenticity_token and personal info 1`] = `
FormData {
  Symbol(state): [
    {
      "name": "authenticity_token",
      "value": "the_token",
    },
    {
      "name": "button",
      "value": "",
    },
    {
      "name": "_method",
      "value": "patch",
    },
    {
      "name": "course[gender]",
      "value": "male",
    },
    {
      "name": "course[last_name]",
      "value": "DUPONT",
    },
    {
      "name": "course[first_name]",
      "value": "Marc",
    },
    {
      "name": "course[birthdate(3i)]",
      "value": "1",
    },
    {
      "name": "course[birthdate(2i)]",
      "value": "1",
    },
    {
      "name": "course[birthdate(1i)]",
      "value": "1990",
    },
    {
      "name": "course[email]",
      "value": "test@test.com",
    },
  ],
}
`;

exports[`The PPSApiFormDataGenerator class The generatePrecautionsFormData method should return a formData containing the default values with provided authenticity_token and personal info 1`] = `
FormData {
  Symbol(state): [
    {
      "name": "authenticity_token",
      "value": "the_token",
    },
    {
      "name": "button",
      "value": "",
    },
    {
      "name": "_method",
      "value": "patch",
    },
    {
      "name": "course[precautions_video]",
      "value": "1",
    },
    {
      "name": "course[precautions_checkbox]",
      "value": "1",
    },
  ],
}
`;

exports[`The PPSApiFormDataGenerator class The generateRiskFactorsFormData method should return a formData containing the default values with provided authenticity_token and personal info 1`] = `
FormData {
  Symbol(state): [
    {
      "name": "authenticity_token",
      "value": "the_token",
    },
    {
      "name": "button",
      "value": "",
    },
    {
      "name": "_method",
      "value": "patch",
    },
    {
      "name": "course[risk_factors_video]",
      "value": "1",
    },
    {
      "name": "course[risk_factors_checkbox]",
      "value": "1",
    },
  ],
}
`;
